src/utils.ts

```diff
@@ -294,7 +310,7 @@ export function areSetsEqual(
 
         b.forEach((bValue) => {
           if (!isValueEqual) {
-            isValueEqual = isEqual(aValue, bValue, meta);
+            isValueEqual = isEqual(aValue, bValue, undefined, a, b, meta);
```

Comment:
Hmm ... I realize the issue here, but having an `undefined` key feels unusual. In `Set.forEach()`, the `key` is the same as the `value`, and since `b` is the driver here perhaps pass through from the iteration:

```ts
b.forEach((bValue, bKey) => {
  if (!isValueEqual) {
    isValueEqual = isEqual(aValue, bValue, bKey, a, b, meta);
```

Another possibility would be to keep a local counter, and pass it through:

```ts
let iterationIndex = 0;

b.forEach((bValue) => {
  if (!isValueEqual) {
    isValueEqual = isEqual(aValue, bValue, iterationIndex++, a, b, meta);
```

This may be helpful for order-based comparisons in a custom comparator. Thoughts?

Reply:
I think your second proposition with an `iterationIndex` is a good idea. The only thing is it could break if parentA or B is access with index (parentA[index]), but since the value is already provided, why would it be ? We could provide the parents as arrays, but I think the performance drop that it would cause is not worth it.

Reply:
Yup, I understand the dilemma, but doesn't the same issue exist with `undefined`? `parentA[undefined]` would break in the same way.

Reply:
Right, I didn't think of `parentA.get(undefined)`.
I'll add the `iterationIndex` !